server/routes/reports/prisoners/remove/removePrisoner.test.ts

```diff
@@ -0,0 +1,61 @@
+import request from 'supertest'
+import type { Express } from 'express'
+import { convertReportWithDetailsDates } from '../../../../data/incidentReportingApiUtils'
+import { mockReport } from '../../../../data/testData/incidentReporting'
+import { now } from '../../../../testutils/fakeClock'
+import { IncidentReportingApi } from '../../../../data/incidentReportingApi'
+import UserService from '../../../../services/userService'
+import { appWithAllRoutes } from '../../../testutils/appSetup'
+import type { User } from '../../../../data/manageUsersApiClient'
+
+jest.mock('../../../../data/incidentReportingApi')
+jest.mock('../../../../services/userService')
+
+let app: Express
+let incidentReportingApi: jest.Mocked<IncidentReportingApi>
+let userService: jest.Mocked<UserService>
+
+beforeEach(() => {
+  userService = UserService.prototype as jest.Mocked<UserService>
+  app = appWithAllRoutes({ services: { userService } })
+  incidentReportingApi = IncidentReportingApi.prototype as jest.Mocked<IncidentReportingApi>
+
+  const users: Record<string, User> = {
+    user1: {
+      username: 'user1',
+      name: 'John Smith',
+    },
+    lev79n: {
+      username: 'lev79n',
+      name: 'Barry Harrison',
+    },
+  }
+  userService.getUsers.mockResolvedValueOnce(users)
+})
+
+afterEach(() => {
+  jest.resetAllMocks()
+})
+
+describe('GET remove prisoner confirmation page', () => {
+  let viewReportUrl: string
+
+  it('should render title question and options to confirm or deny removal', () => {
+    const mockedReport = convertReportWithDetailsDates(
+      mockReport({ reportReference: '6543', reportDateAndTime: now, withDetails: true }),
+    )
+    incidentReportingApi.getReportWithDetailsById.mockResolvedValueOnce(mockedReport)
+    viewReportUrl = `/reports/${mockedReport.id}/prisoners/remove/1`
+    return request(app)
+      .get(viewReportUrl)
+      .expect('Content-Type', /html/)
+      .expect(res => {
+        expect(res.text).toContain('Are you sure you want to remove A1111AA: Andrew Arnold?')
+        expect(res.text).toContain('Yes')
+        expect(res.text).toContain('No')
+        expect(res.text).toContain('Continue')
+        expect(res.text).toContain('Save and exit')
+        expect(incidentReportingApi.getReportWithDetailsById).toHaveBeenCalledTimes(1)
```

Comment:
we should assert that `deleteFromReport` is called with the correct value.

you'd need a little bit of mock setup, though. check out the use of `incidentReportingRelatedObjects ` here:
https://github.com/ministryofjustice/hmpps-incident-reporting/pull/353/commits/dddc665c90d554226066199e07a563229277fdd1#diff-4b826d58631e95f8b773a3121046c0882e3b68fc65609f26020d1401c61cf37bR35-R37